lib/oli/delivery/sections.ex

```diff
@@ -2175,4 +2174,94 @@ defmodule Oli.Delivery.Sections do
     ] ++
       get_flatten_hierarchy(rest, resources)
   end
+
+  defp get_parent_project_survey(section_id) do
```

Comment:
Retrieving the correct revision for the required survey for a course section should be simpler than this code.  Instead, we only need to do two things:

1. Check to see if `required_survey_resource_id` on the `Section` is not `nil`.  If it is `nil`, there is no survey required.
2. With the value of `required_survey_resource_id`, make a call to `Oli.Publishing.DeliveryResolver.from_resource_id`, which resolves the correct revision of the resource for that section.